src/coreComponents/physicsSolvers/surfaceGeneration/SurfaceGenerator.cpp

```diff
@@ -3662,11 +3663,15 @@ int SurfaceGenerator::calculateElementForcesOnEdge( DomainPartition const & doma
 
   arrayView2d< real64 const, nodes::REFERENCE_POSITION_USD > const & X = nodeManager.referencePosition();
 
-  ConstitutiveManager const & cm = domain.getConstitutiveManager();
-  ConstitutiveBase const & solid = cm.getConstitutiveRelation< ConstitutiveBase >( m_solidMaterialNames[0] );
-  m_solidMaterialFullIndex = solid.getIndexInParent();
-
   ConstitutiveManager const & constitutiveManager = domain.getConstitutiveManager();
+  m_solidMaterialFullIndex.resize( elementManager.numRegions() );
+  elementManager.forElementRegionsComplete< CellElementRegion >( [&]( localIndex regionIndex,
+                                                                      CellElementRegion const & region )
+  {
+    string const & solidMaterialName = region.getSubRegion( 0 ).getReference< string >( viewKeyStruct::solidMaterialNameString() );
```

Comment:
I'm a bit surprised by this `getSubRegion( 0 )`.
I suppose that we take the first one because they're all the same across the `sub regions`.
So why don't we store this business information in the `region`?

Reply:
We should be storing a lot of things in Region. Follow up PR.